src/components/wizard_steps/InstalationProgress.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <p>Instalation:</p>
+  <p v-if="all_settings">The following versions will be installed:<span>{{ idf_versions.join(", ") }}</span></p>
+  <n-button @click="startInstalation()">Start Instalation</n-button>
+  <hr>
+  <div v-if="tools">
+    <p>Tools:</p>
+    <table>
+      <tr>
+        <th>Name</th>
+        <th>Downloaded</th>
+        <th>Extracted</th>
+        <th>Finished</th>
+        <th>Error</th>
+      </tr>
+      <tr v-for="tool in tools">
+        <td>
+          {{ tool.name }}
+        </td>
+        <td>
+          {{ tool.downloaded ? 'yes' : 'no' }}
+        </td>
+        <td>
+          {{ tool.extracted ? 'yes' : 'no' }}
+        </td>
+        <td>
+          {{ tool.finished ? 'yes' : 'no' }}
+        </td>
+        <td>
+          {{ tool.error ? 'yes' : 'no' }}
+        </td>
+      </tr>
+    </table>
+  </div>
+  <hr>
+</template>
+
+<script>
+import { ref } from "vue";
+import { invoke } from "@tauri-apps/api/core";
+import { NButton, NSpin } from 'naive-ui'
+import loading from "naive-ui/es/_internal/loading";
+import { listen } from '@tauri-apps/api/event'
+
+
+export default {
+  name: 'InstalationProgress',
+  props: {
+    nextstep: Function
+  },
+  components: { NButton, NSpin },
+
+  data: () => ({
+    os: undefined,
+    all_settings: undefined,
+    loading: true,
+    tools: {},
+  }),
+  methods: {
+    startInstalation: async () => {
+      const _ = invoke("start_installation", {});
+      console.log('start instalation ended!!!')
+      return false;
+    },
+    startListening() {
+      console.log('Listening for tools messages...');
+      listen('tools-message', (event) => {
+        // console.log('Received progress message:', event);
+        switch (event.payload.action) {
```

Comment:
Hopefully event is only received as well formed so payload exists. I suggest Logging here will be beneficial for analysis